unit_tests/test_rest_asset_client.py

```diff
@@ -0,0 +1,294 @@
+# pylint: disable=missing-function-docstring, no-member, too-many-public-methods
+# pylint: disable=too-many-arguments
+import uuid
+from unittest.mock import patch, MagicMock
+
+import pytest
+from gsy_e_sdk.constants import MAX_WORKER_THREADS
+
+from gsy_e_sdk.clients.rest_asset_client import RestAssetClient
+
+TEST_ASSET_UUID = str(uuid.uuid4())
+TEST_SIMULATION_ID = str(uuid.uuid4())
+TEST_AGGREGATOR_UUID = str(uuid.uuid4())
+TEST_TRANSACTION_ID = str(uuid.uuid4())
+
+TEST_DOMAIN_NAME = "test.domain@name.com"
+TEST_WEBSOCKETS_DOMAIN_NAME = "wss:test.domain@name.com"
+
+TEST_COMMAND_RESPONSE = {"command": "register",
+                         "transaction_id": TEST_TRANSACTION_ID,
+                         "registered": True
+                         }
+
+
+@pytest.fixture(name="mock_connections", autouse=True)
+def fixture_mock_connections(mocker):
+    # gsy_framework/client_connections/utils.py
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.retrieve_jwt_key_from_server")
+    mocker.patch("gsy_framework.client_connections.utils.RepeatingTimer")
+
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.WebsocketThread")
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.ThreadPoolExecutor")
+
+
+@pytest.fixture(name="mock_environment_use_functions")
+def fixture_mock_env_use_functions(mocker):
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.simulation_id_from_env",
+                 return_value=TEST_SIMULATION_ID)
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.domain_name_from_env",
+                 return_value=TEST_DOMAIN_NAME)
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.websocket_domain_name_from_env",
+                 return_value=TEST_WEBSOCKETS_DOMAIN_NAME)
+
+
+@pytest.fixture(name="mock_transaction_id")
+def mock_transaction_id_and_timeout_blocking(mocker):
+    mocker.patch("gsy_framework.client_connections.utils.uuid.uuid4",
+                 return_value=TEST_TRANSACTION_ID)
+
+
+@pytest.fixture(name="client")
+def fixture_rest_asset_client(mock_environment_use_functions):  # pylint: disable=unused-argument
```

Comment:
why not setting random values for simulation_id, domain_name and websocket_domain and omitting passing the mock_environment_use_functions here?

Reply:
The values for those attributes still being random. It is just that they are set up directly from the mocked "read_from_env" functions. 
However I see maybe what catch your eyes, being that it would not be necessary to apply the mock to this fixture if the parameters were explicitly set up.